Loan Feature

FrankKleiton opened the pull request:
Fixes #6 

** Description **
This Pull Request intent to cover the application material's loan feature. In addition to that it was made some refactoring to improve the readability of the code. It was implemented considering the current state of  the migrations and not future refactorings.

** What was implemented **
- [ x ] Loan Migration
- [ x ] Models Relationships
- [ x ] Requests
- [ x ] Middleware
- [ x ] Resource Controller
- [ x ] Commentarys
- [ x ] Refactorings

FrankKleiton committed:
:art: Create Loan

It was implemented the store method of the loan, due to the abscence of LostMaterials, it was used a mock function called isLost for make possible develop this feature simutaneosly with the feature lost materials. This is a rudimentary implementations and will be refactored.

FrankKleiton committed:
:package: Installing dependencys

FrankKleiton committed:
:card_file_box: Improving the migrations structure

FrankKleiton committed:
:art: Adding relationships and mock function to Material model

FrankKleiton committed:
:art: Improving the Loan functionality

FrankKleiton committed:
:recycle: Adding commentarys and refactoring

FrankKleiton committed:
:recycle: Usign Request Forms

FrankKleiton committed:
:twisted_rightwards_arrows: Resoving conflicts by incorporing both segestions

FrankKleiton committed:
:recycle: Adding comentarys

FrankKleiton committed:
:recycle: Improving the create loan feature

FrankKleiton committed:
:recycle: Hidden related tables data

FrankKleiton committed:
:art: Saving current user in the material's store

FrankKleiton committed:
:art: Loan Index, Show and Update Actions Implemented

FrankKleiton committed:
:recycle: Removing Loan's Refresh Call in Action Update

FrankKleiton committed:
:recycle: Improving Validation Structure

FrankKleiton committed:
:art: Soft Deleting Loans

FrankKleiton committed:
:art: Adding New Validation to Destroy Action in Loan

FrankKleiton committed:
:art: Improving The Datetime Format on Serialization

FrankKleiton committed:
:recycle: Finishing Loan Controller

hdev14 reviewed (changes requested):
I also have see that you associated the user into a material. In the new diagram doesn't exist this relationship, but as we lets remake the migrations, it is ok!

FrankKleiton committed:
:ok_hand: Removing the loaned column from loan

The UpdateLoan Request was deleted and the update and destroy action from LoanController were updated.

hdev14 reviewed (approved):
:coffee:
